Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -59,19 +104,96 @@ class PreplannedMapModel: ObservableObject, Identifiable {
+    /// - Precondition: `job.status == .succeeded || job.status == .failed`
+    private static func notifyJobCompleted(job: DownloadPreplannedOfflineMapJob) async throws {
+        precondition(job.status == .succeeded || job.status == .failed)
+        guard
+            let preplannedMapArea = job.parameters.preplannedMapArea,
+            let id = preplannedMapArea.id
+        else { return }
+        
+        let content = UNMutableNotificationContent()
+        content.sound = UNNotificationSound.default
+        
+        let jobStatus = job.status == .succeeded ? "Succeeded" : "Failed"
+        
+        content.title = "Download \(jobStatus)"
+        content.body = "The job for \(preplannedMapArea.title) has \(jobStatus.lowercased())."
+        
+        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
+        let identifier = id.rawValue
+        let request = UNNotificationRequest(identifier: identifier, content: content, trigger: trigger)
+        
+        try await UNUserNotificationCenter.current().add(request)
+    }
+    
+    /// Downloads the preplanned map area.
+    /// - Precondition: `canDownload`
+    func downloadPreplannedMapArea() async {
+        precondition(status.allowsDownload)
+        status = .downloading
+        
+        do {
+            let parameters = try await preplannedMapArea.makeParameters(using: offlineMapTask)
+            let mmpkDirectory = FileManager.default.preplannedDirectory(
+                forPortalItemID: portalItemID,
+                preplannedMapAreaID: preplannedMapAreaID
+            )
+            try FileManager.default.createDirectory(at: mmpkDirectory, withIntermediateDirectories: true)
+            
+            // Create the download preplanned offline map job.
+            let job = offlineMapTask.makeDownloadPreplannedOfflineMapJob(
+                parameters: parameters,
+                downloadDirectory: mmpkDirectory
+            )
+            JobManager.shared.jobs.append(job)
+            startAndObserveJob(job)
+        } catch {
+            status = .downloadFailure(error)
+        }
+    }
+    
+    /// Sets the job property of this instance, starts the job, observes it, and
+    /// when it's done, updates the status, removes the job from the job manager,
+    /// and fires a user notification.
+    private func startAndObserveJob(_ job: DownloadPreplannedOfflineMapJob) {
+        self.job = job
+        job.start()
+        status = .downloading
+        Task { @MainActor in
+            let result = await job.result
+            updateDownloadStatus(for: result)
+            mobileMapPackage = try? result.map { $0.mobileMapPackage }.get()
+            JobManager.shared.jobs.removeAll { $0 === job }
+            if showsUserNotificationOnCompletion && (job.status == .succeeded || job.status == .failed) {
+                try? await Self.notifyJobCompleted(job: job)
+            }
         }
```

Comment:
A few questions regarding the `Task`

- Since the `PreplannedMapModel` is annotated with `@MainActor` as a whole, I think each member method inherits the actor-context to be main actor. Since `Task` block implicitly inherits the actor-context, should be no need to write out `@MainActor in` again…?

```suggestion
        Task {
            let result = await job.result
            updateDownloadStatus(for: result)
            mobileMapPackage = try? result.map { $0.mobileMapPackage }.get()
            JobManager.shared.jobs.removeAll { $0 === job }
            if showsUserNotificationOnCompletion && (job.status == .succeeded || job.status == .failed) {
                try? await Self.notifyJobCompleted(job: job)
            }
        }
```
- Say we have a long running `DownloadPreplannedOfflineMapJob` instance that takes 1 minute. Because the unstructured `Task` block implicitly capture `self`, the data model instance won't go away until the job's reference is released (both by the job manager and the data model, presumably). In this case, is using a unstructured task here OK?
- Shall we set `self.job = nil` after the job completes?

Reply:
I verified that the Main Actor annotation is not needed here because the actor context is inherited.